types/mock-property/index.d.ts

```diff
@@ -0,0 +1,23 @@
+// Type definitions for mock-property 1.0
+// Project: https://github.com/ljharb/mock-property#readme
+// Definitions by: Jordan Harband <https://github.com/ljharb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type RestoreFunction = () => void;
+
+declare function mockProperty(
+    obj: object,
+    prop: PropertyKey,
+    options?: (
+        | { delete: true }
+        | (
+            { nonEnumerable?: boolean, delete?: false }
+            & (
+                | { nonWritable?: boolean, value?: unknown }
+                | { get?: (() => unknown) | undefined, set?: ((v: unknown) => void) | undefined }
```

Comment:
These `unknown`s make me feel like this function could really be generic and more type safe, but, it's probably fine for now.